client/login/wp-login/index.jsx

```diff
@@ -282,7 +290,101 @@ export class Login extends Component {
 		);
 	}
 
-	renderContent() {
+	recordResetPasswordLinkClick = () => {
+		this.props.recordTracksEvent( 'calypso_login_reset_password_link_click' );
+	};
+
+	getLostPasswordLink() {
+		if ( this.props.twoFactorAuthType || this.props.privateSite ) {
+			return null;
+		}
+
+		let lostPasswordUrl = lostPassword( { locale: this.props.locale } );
+
+		// If we got here coming from Jetpack Cloud login page, we want to go back
+		// to it after we finish the process
+		if ( isJetpackCloudOAuth2Client( this.props.oauth2Client ) ) {
+			const currentUrl = new URL( window.location.href );
+			currentUrl.searchParams.append( 'lostpassword_flow', true );
+			const queryArgs = {
+				redirect_to: currentUrl.toString(),
+
+				// This parameter tells WPCOM that we are coming from Jetpack.com,
+				// so it can present the user a Lost password page that works in
+				// the context of Jetpack.com.
+				client_id: this.props.oauth2Client.id,
+			};
+			lostPasswordUrl = addQueryArgs( queryArgs, lostPasswordUrl );
+		}
+
+		return (
+			<a
+				href={ lostPasswordUrl }
+				key="lost-password-link"
+				className="login__lost-password-link"
+				onClick={ this.recordResetPasswordLinkClick }
+				rel="external"
+			>
+				{ this.props.translate( 'Lost your password?' ) }
+			</a>
+		);
+	}
+
+	renderSignUpLink( signupLinkText ) {
+		// Taken from client/layout/masterbar/logged-out.jsx
+		const {
+			currentRoute,
+			isP2Login,
+			locale,
+			oauth2Client,
+			pathname,
+			query,
+			translate,
+			usernameOrEmail,
+		} = this.props;
+
+		// use '?signup_url' if explicitly passed as URL query param
+		const signupUrl = this.props.signupUrl
+			? window.location.origin + pathWithLeadingSlash( this.props.signupUrl )
+			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname );
+
+		if ( isJetpackCloudOAuth2Client( oauth2Client ) && '/log-in/authenticator' !== currentRoute ) {
```

Comment:
Maybe we could move this check and the next one up? To avoid this unnecessary check?
```js
const signupUrl = this.props.signupUrl
			? window.location.origin + pathWithLeadingSlash( this.props.signupUrl )
			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname );
``` 

Line 351 and 355 before 347

Reply:
Makes totally sense 👍